lib/board.ex

```diff
@@ -12,4 +12,53 @@ defmodule Board do
   def place_a_symbol(board, player_move, symbol) do
     Map.replace(board, String.to_integer(player_move), symbol)
   end
+
+  @spec has_player_won(map()) :: boolean()
+  def has_player_won(board) do
+    Enum.any?([get_all_rows_status(board), get_all_columns_status(board), get_both_diagonals_status(board)])
+  end
+
+  @spec get_board_dimensions(map()) :: integer()
+  def get_board_dimensions(board) do
+    trunc(:math.sqrt(Enum.count(board)))
+  end
+
+  @spec get_all_rows_status(map()) :: boolean()
+  defp get_all_rows_status(board) do
+    board_dimensions = get_board_dimensions(board)
+    board
+    |> Map.values()
+    |> Enum.chunk_every(board_dimensions)
+    |> Enum.map(fn rows -> Enum.uniq(rows) end)
+    |> Enum.any?(fn win -> Enum.count(win) == 1 end)
+  end
+
+  @spec get_all_columns_status(map()) :: boolean()
+  defp get_all_columns_status(board) do
+    board_dimensions = get_board_dimensions(board)
+    board
+    |> Map.values()
+    |> Enum.chunk_every(board_dimensions)
+    |> Enum.zip()
+    |> Enum.map(fn columns -> Enum.uniq(Tuple.to_list(columns)) end)
+    |> Enum.any?(fn win -> Enum.count(win) == 1 end)
+  end
+
+  @spec get_both_diagonals_status(map()) :: boolean()
+  defp get_both_diagonals_status(board) do
+    board_dimensions = get_board_dimensions(board)
+    board
+    |> Map.values()
+    |> Enum.chunk_every(board_dimensions)
+    |> get_diagonal_values(board_dimensions)
+    |> Enum.map(fn columns -> Enum.uniq(columns) end)
```

Comment:
```suggestion
    |> Enum.map(&Enum.uniq/1)
```

Code 🏌🏽‍♂️ – whenever you are wrapping a function with an anonymous function like this you can use the `&fn/arity` shorthand.

Reply:
Makes sense to me, this is awesome to know as well! You can't do this for line 55 though right? Since its performing an operation and not just wrapping a function?

Reply:
Right, you're doing an extra comparison on 55 instead of just wrapping another function.